### DISCUSSION TITLE
Notes May 11th, 2022

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre#28] [Technical Steering Committee Meetings]
## Organizational Questions
* @ntadej: Containers permissions for native builds - @wipfli to check permissions
* @nyurik: Are we OK to have an odd number of Governing Board members?
Having 5 is fine
Board Terms (2 year period? - 1 year overlap). Check Marcel notes: @mojodna can put us in touch with David Nalley from Apache to talk about governance, and the Jupiter foundation has a good model. Check also OSGEO presidency.
* @nyurik: New repos announcement
New [repo checklist](https://github.com/maplibre/maplibre/blob/main/.github/ISSUE_TEMPLATE/new-repo-checklist.md)
[maplibre-rs](https://github.com/maplibre/maplibre/issues/24) - Experimental MapLibre re-implementation in Rust that can compile to WebAssembly and native, and uses WebGPU and/or other GPU UI tech
[Mbtileserve-rs](https://github.com/maplibre/maplibre/issues/21) - a simple .mbtiles tile server
Going forward, we [need to discuss](https://github.com/maplibre/maplibre/discussions/27) the scope of MapLibre, and solicit community feedback using.
Proposal to implement Rust bindings for maplibre-native, stored in the same repo as native. Discussed on Slack.
* @HarelM: Suggested MapLibre for Appwrite.io open-source funds - https://appwrite.io/oss-fund-announcement 

## MapLibre GL JS Technical Questions
* @prozessor13 3D merge roadmap. Need some more tests in principle, see https://github.com/maplibre/maplibre-gl-js/pull/1189, but we should not take the pull request hostage because of these. So let's merge the Terrain3D branch!

## MapLibre Native Technical Questions:
* @ntadej: Qt setup almost done ([Qt build improvements and documentation](https://github.com/maplibre/maplibre-gl-native/pull/277), [[qt] Initial import of Qt QML plugin](https://github.com/maplibre/maplibre-gl-native/pull/263))
* @ntadej: Regression: [Highway shield (glyph?) text not centered.](https://github.com/maplibre/maplibre-gl-native/issues/279)
* @ntadej: Build cleanup, documentation and modernisation:
Should we make a stable branch and more experimental devel branch? Needs a plan:
Merging of metal branch.
Proposal: not yet. Amazon to work on plan on proper Metal implementation (without Angle emulation layer)
C++17 migration
Cleanup of dependencies
Potential renaming of some SDKs/bindings to MapLibre
* @HarelM Shaders in MapLibre GL JS are also used in the native repo. What is the process if we update the shaders on the GL JS side? What tests do we need to run on the native side? @birkskyum started a [typescript shaders branch](https://github.com/maplibre/maplibre-gl-native/tree/generate-shaders-using-typescript) in the native repo.

## Attendees
@wipfli
@HarelM
@ntadej
@roblabs
@klokan 
@nyurik
@prozessor13
@Joxit
@mojodna

We are happy for anyone's feedback!

**Comment by lseelenbinder** [comment]
@nyurik We have Rust bindings for Mapbox Native GL which we can look into porting and partially open-sourcing, once this is fixed: https://github.com/maplibre/maplibre-gl-native/issues/279.

**Comment by lseelenbinder** [comment]
> Are we OK to have an odd number of Governing Board members?

I think we _should_ have an odd number. It allows us to break any ties in voting without any issue.
